nixos/modules/services/misc/gitea.nix

```diff
@@ -163,6 +163,58 @@ in
         };
       };
 
+      captcha = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Enables Gitea to display a CAPTCHA challenge on registration.
+          '';
+        };
+
+        secretFile = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "/var/lib/secrets/gitea/captcha_secret";
+          description = "Path to a file containing the CAPTCHA secret key.";
+        };
+
+        siteKey = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "my_site_key";
```

Comment:
Has this to be considered a secret?

Reply:
No. The site key is typically provided to the client through the webpage to make CAPTCHA queries, so it is expected to be publicly accessible. https://developers.google.com/recaptcha/docs/v3#automatically_bind_the_challenge_to_a_button